lib/semantics/resolve-labels.cc

```diff
@@ -889,9 +907,16 @@ void CheckLabelDoConstraints(const SourceStmtList &dos,
               SayLabel(label)});
     } else if (!InInclusiveScope(scopes, scope, doTarget.proxyForScope)) {
       // C1133
-      errorHandler.Say(position,
-          parser::MessageFormattedText{
-              "label '%u' is not in scope"_en_US, SayLabel(label)});
+      if (doTarget.legacyAcceptedInDirectSubscope &&
+          ParentScope(scopes, doTarget.proxyForScope) == scope) {
+        errorHandler.Say(doTarget.parserCharBlock,
+            parser::MessageFormattedText{
+                "A DO loop should terminate with an END DO or CONTINUE inside its scope"_en_US});
```

Comment:
Can this case be merged with the succeeding `else if` case rather than replicating the message?

Reply:
Thanks, it saves some lines and the merged conditions remain readable.